Formatter/Pool.php

```diff
@@ -34,15 +34,13 @@ class Pool
     protected $logger;
 
     /**
-     * @param LoggerInterface|null $logger
-     * @param string|null          $defaultFormatter
+     * @param string          $defaultFormatter
+     * @param LoggerInterface $logger
      */
-    public function __construct(LoggerInterface $logger = null, $defaultFormatter = null)
+    public function __construct($defaultFormatter, LoggerInterface $logger)
     {
-        $this->logger = $logger;
-
-        // TODO: This should become a required first parameter when the major version is changed
```

Comment:
@sagikazarmark I think there is no point anymore to switch the arguments if the logger can't be null, am I right?